### ISSUE TITLE
[New build - EXPERIMENTAL] 20w28c: New Rendering Tech.

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2653]
This is the second stage of the new rendering technology. This is a major update, please report any potential regression!

Changes:
- The CPU is now doing a minimum amount of work when a brick is added. 
- Performances should be a bit better on the CPU side.

Bugfixes:
This should fix the dreaded brick add lag issue on large build.

**Comment by tsunamayo** [comment]
Build is up!
As this quite a big update I am pushing this on Experimental, please use the beta tab in steam to subscribe to this build. With all the issues on AMD and co, it is a lot safer that way! If it works for you, please answer to this post and tell me your GFX card. Thanks a lot all!

**Comment by tsunamayo** [comment]
Okay while doing further testing with large build I have noticed two issues:
 - Some block are disappearing on the edge of the screen when the camera is moving fast
- When deleting a lot of blocks there is one frame where previously deleted block becomes visible!
I am checking both issues.

**Comment by ZachZent** [comment]
-  The CPU is now doing a minimum amount of work when a brick is added.
    Performances should be a bit better on the CPU side.

Do you mean GPU? SEVO CPU usage is extremely low while GPU use is high.

**Comment by ultrak2k** [comment]
Yeah, I've heard people say that the game uses less CPU then terraria, is this a GPU thing?

**Comment by chacalik** [comment]
I'm not sure why, but when I approach my ship, a dark shadow begins to cover the whole surface I'm pointing at and neither flashlight work on this surface.
![image](https://user-images.githubusercontent.com/4794243/87011540-70d19b80-c19e-11ea-8f0e-0c60fa9c7332.png)
![image](https://user-images.githubusercontent.com/4794243/87011579-7d55f400-c19e-11ea-9763-3d07b944e752.png)
![image](https://user-images.githubusercontent.com/4794243/87011603-85159880-c19e-11ea-87e8-da19c30ebab9.png)

**Comment by alxnns1** [comment]
On creating a new world I see the below screen:
![20200709093630_1](https://user-images.githubusercontent.com/485571/87017390-d1290300-c1c7-11ea-82b1-35c0733ddcd3.jpg)

GPU: GTX 1050 Ti (laptop)

Dunno if you also want out output logs:
[output_log.txt](https://github.com/tsunamayo/Starship-EVO/files/4895515/output_log.txt)

**Comment by tsunamayo** [comment]
@alxnns1 and you have no issue on loading an existing save? Thanks

**Comment by alxnns1** [comment]
When I load an existing save I get the same

**Comment by tsunamayo** [comment]
@ZachZent it really depends on your config and the scene, but I can tell you that I really need some CPU improvements as gameplay and NPCs stuff really require some work on the CPU side... If you are checking the perf while doing nothing (or just building) then yes the CPU should not be the limiting factor, otherwise that would be pretty bad...

**Comment by tsunamayo** [comment]
@chacalik not sure to understand what is going on, is it simply normal shadow? Shadow will kick-off at 200m only, so it might just be that.

**Comment by Kaiser-Indrasil** [comment]
I'm getting the exact same bug @alxnns1 got. Plus, the loading time seems much longer.
When creating a new world, the loading artwork persists after the world been loaded. I can hear the sounds in the game but the loading screen obstructs all the view.
...and the game crashed itself along with several other programs.

**Comment by tsunamayo** [comment]
@alxnns1 @Kaiser-Indrasil what are your GFX cards? Plus a log for kaiser would be great to investigate. Error in the case of alxnns seems to be related to the outline post process (which I had to tweak also)

**Comment by alxnns1** [comment]
GPU: GTX 150 Ti (laptop)

and just in case
CPU: i7-8750H

**Comment by tsunamayo** [comment]
ah sorry mate you already told me ;)

**Comment by Kaiser-Indrasil** [comment]
Sure, here are the logs and the crash dump:

[error.log](https://github.com/tsunamayo/Starship-EVO/files/4896966/error.log)
[output_log.txt](https://github.com/tsunamayo/Starship-EVO/files/4896967/output_log.txt)
[crash.zip](https://github.com/tsunamayo/Starship-EVO/files/4896969/crash.zip)

My GPU is 2080Ti

**Comment by tsunamayo** [comment]
@Kaiser-Indrasil lucky guy! btw as I never managed to get a proper crash and the unity doc seems outdated where did you find the crash dump and error.log?

**Comment by tsunamayo** [comment]
Ah I think I got that error fixed, mine are also fixed so I am pushing a new build.
@ZachZent also I forgot to add one thing, there will be a big GPU improvement for large ship because of shadows. Before shadow visibility were computed per ship, now is its per brick. It as a huge bottleneck for large ship, as you were submitting a lot of shadow work that was not shown at the end. With a bit more work and tweak I could add a higher shadow quality setting for better GPU...
Cheers

**Comment by Kaiser-Indrasil** [comment]
Oh man, am I glad that you decided to delegate the new version to the experimental version... there's so many bugs:
- #2636 still persists
- 10FPS when the ship is near the torus
- 50% fps drop (from 100 down to 50) when Occlusion Culling is enabled. That's initially. After letting the game run for a few minutes, it oscillates between 60 and 80 fps.
- the Occlusion Culling-related bug
https://i.imgur.com/EtZppk9.png
There is no hole in the ship's hull but the occlusion tech doesn't render the armour at some degrees.

**Comment by Kaiser-Indrasil** [comment]
I found it in the %appdata%\Local\Temp\Moonfire Entertainment\Starship EVO\Crashes\Crash_2020-07-09_125014691 folder. That's where the instruction at the end of the log file pointed me to.
